scripts/serial-test.sh

```diff
@@ -1,36 +1,21 @@
 #!/bin/bash
-set -e # exit on failure of any "simple" command (excludes &&, ||, or | chains)
-# prepare environment
+set -eo pipefail
+[[ -f build.tar.gz ]] && tar -xzf build.tar.gz
+cd ./build
+echo "[Killing old MongoDB]"
+$(pgrep mongod | xargs kill -9) || true
+echo "[Starting MongoDB]"
 PATH=$PATH:~/opt/mongodb/bin
-echo "Extracting build directory..."
-[[ -z "${1}" ]] && tar -zxf build.tar.gz || tar -xzf $1
-echo "Starting MongoDB..."
-mongod --fork --dbpath ~/data/mongodb -f ~/etc/mongod.conf --logpath "$(pwd)"/mongod.log
-cd /data/job/build
-# run tests
-echo "Running tests..."
-TEST_COUNT=$(ctest -N -L nonparallelizable_tests | grep -i 'Total Tests: ' | cut -d ':' -f 2 | awk '{print $1}')
-[[ $TEST_COUNT > 0 ]] && echo "$TEST_COUNT tests found." || (echo "ERROR: No tests registered with ctest! Exiting..." && exit 1)
+mongod --fork --dbpath ~/data/mongodb -f ~/etc/mongod.conf --logpath ~/var/log/mongodb/mongod.log
```

Comment:
If you don't move it before uploading the artifact ends up being the absolute path to the file /Users/blah/var/log/mongodb/mongod.log in buildkite. If we try and download it, we have to specify that full path/name for the artifact. 

The build directory is always writable, though I've added -f for some mv assurance that it will overwrite. We added an ENV that forces the "working directory" for buildkite to clean itself before doing the git clone/checkout, so that should definitely prevent any sort of collisions.

Reply:
You're right. I wanted to keep things consistent with the build instructions, but I realize now that's futile.